### PR TITLE
Alembic export

### PR DESCRIPTION
**Pull request by cpichard**
This pull request adds alembic support in openMVG, providing the ability to export points with colors and cameras with their image plane. 
I am not sure which branch we should merge with, please don't hesitate to switch branch if needed.

Cyril

### CODE REVIEW

**Comment by simogasp** [comment]
Hi @cpichard , 
sorry there is a bit of a mess now with the Alembic support. I actually refactored the code and added Import/Export class inside the library (as we did for cameraLocalization and mockUp) If you want to take a look
https://github.com/poparteu/openMVG/tree/voctree_localizer/src/openMVG/dataio
And then I use them for the exporter.
For what i see the main difference with your version is that you added the colored cloud and the image plane. But I think it can be easily integrated in the class I made (adding a new method and refactoring a bit i guess). What do you think? 
Also @fabiencastan @caymard are about to work on that
S.

**Comment by cpichard** [comment]
Hi Simone,
sure, this would be more convenient to have this code inside the library, do you want to merge it yourself in your branch ? it's only a matter of several lines. Let me know !

C

File: src/software/SfM/main_openMVG2ABC.cpp [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        const float dx = cam->principal_point()(0);
+        const float dy = cam->principal_point()(1);
+        // Use a common sensor width as we don't have this information at this point
+        // We chose a full frame 24x36 camera
+        const float sensorWidth = 36.0; // 36mm per default
+        const float sensorHeight = sensorWidth*sensorHeightPix/sensorWidthPix;
+        const float focalLength = sensorWidth*focalLengthPix/sensorWidthPix;
+        // Following values are in cm, hence the 0.1 multiplier
+        const float hoffset = 0.1*sensorWidth*(0.5-dx/imgWidth);
+        const float voffset = 0.1*sensorHeight*(dy/imgHeight-0.5)*sensorHeightPix/sensorWidthPix;
```
Comment (caymard) [inline src/software/SfM/main_openMVG2ABC.cpp]: Why do you multiply by `sensorHeightPix/sensorWidthPix` here?
```
const float voffset = 0.1*sensorHeight*(dy/imgHeight-0.5);
```
Upper there is `sensorHeight = sensorWidth*sensorHeightPix/sensorWidthPix`. So it's like the ratio is applied twice.

**Comment by fabiencastan** [comment]
Code imported manually.

**Pull request closed without merging**